cogs/channel.py

```diff
@@ -0,0 +1,144 @@
+            Parameters:
+                channel_name: the name of the channel
+                category_name: the name of the category
+        """
+        guild = context.guild
+
+        category = await ChannelManager.getCategory(category_name, context)
+
+        return await guild.create_text_channel(channel_name, category=category)
+
+    def makeDict(categories, channels):
+        classDict = {}
+        for channel in channels:
+            for category in categories:
+                if (category in channel):
+                    classDict[channel] = category
+        print(classDict)
+        return classDict
+
+    @commands.command(name="channelparse")
+    @has_permissions(administrator=True)
+    async def channelparse(self, context):
+        """
+        [Optional filename argument] This command sets up channels from a csv
+        """
+
+        channelnames = []
+        categories = []
+
+        # parses command to get message content
+        prefix = config["bot_prefix"]
+        startLen = len(prefix) + len("channelparse")
+        filename = context.message.content[startLen:].strip()
+
+        if(re.search("\W", filename)):
+            await context.send("Please input a filename without special characters or extensions")
+            return
+
+        if (not filename.strip()):
+            await context.send("Please specify a file.")
+            return
+        filename = "resources\\"+filename+".csv"
+        print("filename: ", filename)
+
+        # read and parse the csv
+        with open(filename, newline='') as csvfile:
+            csvreader = csv.reader(csvfile, delimiter=',')
+            for row in csvreader:
+
+                # check if those attributes exist
+                if(row[2].strip() and row[3].strip() and row[19].strip()):
+
+                    classtype = row[2]
+                    # first 3 numbers only (not L1,L2,L3)
+                    classnum = row[3][0:3]
+                    prof = row[19]
+
+                    if (prof != "TBA"):
+                        prof = prof.split()[1]
+
+                    # assemble class and category names
+                    channelname = classtype+"-"+classnum+"-"+prof
+                    categoryname = classtype + "-"+classnum
+
+                    if(categoryname not in categories):
+                        categories.append(categoryname)
+
+                    channelnames.append(channelname)
+            classDict = ChannelManager.makeDict(categories, channelnames)
+            # unable to creat channels
+            for channel in channelnames:
+                if('388' in channel or '571' in channel or '511' in channel):
+                    continue
+                await ChannelManager.createChannel(channel, classDict[channel], context)
+
+        await context.send("Channels Created Successfully")
+
+# And then we finally add the cog to the bot so that it can load, unload, reload and use it's content.
```

Comment:
We probably don't need this comment, but if you want it here that's okay

Reply:
I appreciate it because it lets the user know the program completed fully, instead of it creating only half the channels and erroring without us knowing.